Python/bytecodes.c

```diff
@@ -1158,18 +1158,21 @@ dummy_func(
         }
 
         macro(STORE_SUBSCR_DICT) =
-            _GUARD_NOS_DICT + unused/1 + _STORE_SUBSCR_DICT;
+            _GUARD_NOS_DICT + unused/1 + _STORE_SUBSCR_DICT + POP_TOP;
 
-        op(_STORE_SUBSCR_DICT, (value, dict_st, sub -- )) {
+        op(_STORE_SUBSCR_DICT, (value, dict_st, sub -- st)) {
             PyObject *dict = PyStackRef_AsPyObjectBorrow(dict_st);
 
             assert(PyDict_CheckExact(dict));
             STAT_INC(STORE_SUBSCR, hit);
             int err = _PyDict_SetItem_Take2((PyDictObject *)dict,
                                             PyStackRef_AsPyObjectSteal(sub),
                                             PyStackRef_AsPyObjectSteal(value));
-            PyStackRef_CLOSE(dict_st);
-            ERROR_IF(err);
+            if (err) {
+                ERROR_NO_POP();
```

Comment:
We can't do this here. The problem is that `_PyDict_SetItem_Take2` has already stolen the references to `sub` and `value`. This means when we jump to error, we decref `sub` and `value` again, causing a double decref.

The fix might be to do
```
if (err) {
    PyStackRef_CLOSE(dict_st);
    ERROR_IF(1);
}
```